tools/clang/tools/dxopt/dxopt.cpp

```diff
@@ -144,14 +184,23 @@ int __cdecl wmain(int argc, const wchar_t **argv_) {
         action = ProgramAction::PrintPassesWithDetails;
       }
       else {
-        // Assume we get a file name followed by possibly -o= then optimizer options.
         action = ProgramAction::RunOptimizer;
-        inFileName = arg;
-        argIdx++;
-        if (argc > 2 && wcsistarts(argv_[argIdx], L"-o=")) {
+        // Next arg does not start with '-' and so is a filename,
+        // or next arg equals '-' so we should read from stdin.
+        if (!wcsistarts(arg, L"-") || isStdIn(arg)) {
```

Comment:
It looks like we support / for pass options but not for dxopt options (e.g. -S). Will change this to assume arg starting with - or / is not a file arg.